Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -494,43 +504,55 @@ - (BOOL)application:(UIApplication *)application
+        [urlString containsString:branchUniversalLinkDomains]) {
+        return YES;
+    }
+    else if ([branchUniversalLinkDomains isKindOfClass:[NSArray class]]) {
+        for (id oneDomain in branchUniversalLinkDomains) {
+            if ([oneDomain isKindOfClass:[NSString class]] && [urlString containsString:oneDomain]) {
+                return YES;
+            }
+        }
+    }
+    
+    NSString *userActivityURL = urlString;
+    NSArray *branchDomains = [NSArray arrayWithObjects:@"bnc.lt", @"app.link", @"test-app.link", nil];
+    for (NSString* domain in branchDomains) {
+        if ([userActivityURL containsString:domain])
+            return YES;
+    }
+    
+    return NO;
+}
+
 - (BOOL)continueUserActivity:(NSUserActivity *)userActivity {
     //check to see if a browser activity needs to be handled
     if ([userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb]) {
-        self.preferenceHelper.universalLinkUrl = [userActivity.webpageURL absoluteString];
-        self.preferenceHelper.shouldWaitForInit = NO;
-        [self initUserSessionAndCallCallback:YES];
-        
-        id branchUniversalLinkDomains = [self.preferenceHelper getBranchUniversalLinkDomains];
-        if ([branchUniversalLinkDomains isKindOfClass:[NSString class]] && [[userActivity.webpageURL absoluteString] containsString:branchUniversalLinkDomains]) {
-            return YES;
-        }
-        else if ([branchUniversalLinkDomains isKindOfClass:[NSArray class]]) {
-            for (id oneDomain in branchUniversalLinkDomains) {
-                if ([oneDomain isKindOfClass:[NSString class]] && [[userActivity.webpageURL absoluteString] containsString:oneDomain]) {
-                    return YES;
-                }
-            }
-        }
-        
-        NSString *userActivityURL = [userActivity.webpageURL absoluteString];
-        NSArray *branchDomains = [NSArray arrayWithObjects:@"bnc.lt", @"app.link", @"test-app.link", nil];
-        for (NSString* domain in branchDomains) {
-            if ([userActivityURL containsString:domain])
-                return YES;
-        }
-        
-        return NO;
+        return [self handleUniversalDeepLink:userActivity.webpageURL fromSelf:NO];
     }
     
     // Check to see if a spotlight activity needs to be handled
-    NSString *spotlightIdentifier = [self.contentDiscoveryManager spotlightIdentifierFromActivity:userActivity];
+    NSString *spotlightIdentifier =
+    [self.contentDiscoveryManager spotlightIdentifierFromActivity:userActivity];
     
     if (spotlightIdentifier) {
         self.preferenceHelper.spotlightIdentifier = spotlightIdentifier;
     }
     else {
-        NSString *nonBranchSpotlightIdentifier = [self.contentDiscoveryManager standardSpotlightIdentifierFromActivity:userActivity];
+        NSString *nonBranchSpotlightIdentifier =
+        [self.contentDiscoveryManager standardSpotlightIdentifierFromActivity:userActivity];
```

Comment:
Can you indent here? So it's
```
NSString *nonBranchSpotlightIdentifier =
   [self.contentDiscoveryManager standardSpotlightIdentifierFromActivity:userActivity];
```

Maybe I can't see the indent because of a tab vs. spaces thing.  I usually indent with spaces.